### DISCUSSION TITLE
Support for sotamat

### DISCUSSION THREAD

**Discussion by aendes** [danleetw/FT8TW#2] [Ideas]
Hello,

thanks for your great work.

Could you please also add the two characters "-" and "/" to the CQ free text?

f71: Conveys arbitrary free text with up to 13 characters
selected from the list
0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ+-./?
plus the blank character. The encoding algorithm is
illustrated by program free_text_to_f71.

https://wsjt.sourceforge.io/FT4_FT8_QEX.pdf

Thanks for your great work!

**Comment by danleetw** [comment]
> Hello,
> 
> thanks for your great work.
> 
> Could you please also add the two characters "-" and "/" to the CQ free text?
> 
> f71: Conveys arbitrary free text with up to 13 characters selected from the list 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ+-./? plus the blank character. The encoding algorithm is illustrated by program free_text_to_f71.
> 
> https://wsjt.sourceforge.io/FT4_FT8_QEX.pdf
> 
> Thanks for your great work!

Thank you for the information! I've learned more about FT8's FreeText feature. I only discovered it earlier when looking through the code and found that long-pressing the CQ button allows access to FreeText input. I'm curious, how did you find out about it? Also, could you share in what scenarios FreeText is typically used?

I have already made modifications. Although I only gained a bit of understanding about Android's input system during this process, customizing input content seems a bit tricky. While I'm not entirely satisfied with the changes, they should be functional now.

The program is currently temporarily placed in ft8tw_0924d6.apk

**Reply by aendes** [reply]
Thanks for your hard work. I can not input the all the allowed characters.

I want to use it with sotamat.com to send FT8 messages to sota

**Reply by danleetw** [reply]
Do you try 0.924d6 or d7 version?

**Comment by danleetw** [comment]
<img width=500 height=600 src=https://github.com/user-attachments/assets/f7436693-f2e4-4f59-b067-e5b1e28c8742>
<!-- ![Screenshot_20241214_220502_FT8TW](https://github.com/user-attachments/assets/f7436693-f2e4-4f59-b067-e5b1e28c8742)>

**Comment by danleetw** [comment]
> Hello,
> 
> thanks for your great work.
> 
> Could you please also add the two characters "-" and "/" to the CQ free text?
> 
> f71: Conveys arbitrary free text with up to 13 characters selected from the list 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ+-./? plus the blank character. The encoding algorithm is illustrated by program free_text_to_f71.
> 
> https://wsjt.sourceforge.io/FT4_FT8_QEX.pdf
> 
> Thanks for your great work!

I added SOTAMAT parameter values in the settings, so when opening FREETEXT, they will be automatically populated. This avoids having to re-enter the SOTAMAT content every time and helps prevent potential input issues. 

Additionally, I’d like to ask: I’ve set up the SOTAMAT Command and SPOT information, but they are not working successfully. Is there any way to test or debug this?

**Reply by aendes** [reply]
There is a kind of Testmode on: https://sotawatch.sota.org.uk/
If the Message contains Test it will be only shown as test message, but not be used by the users

I will test the messaging as soon as the weather gets better;-)

**Reply by danleetw** [reply]
I did it!
![Screenshot_20250109_192555_SOTA Activator](https://github.com/user-attachments/assets/b452122d-0e08-4923-bf50-c9b608480d94)
